srcalc/src/main/java/gov/va/med/srcalc/web/view/EditVariable.java

```diff
@@ -0,0 +1,175 @@
+ * <p>Workflow:</p>
+ * 
+ * <ol>
+ * <li>Construct an instance based on an existing variable.</li>
+ * <li>Present the user with the initial values and the "reference data" such
+ * as {@link #getAllGroups()}.</li>
+ * <li>Have the user update the properties as desired.</li>
+ * <li>Use {@link EditVariableValidator} to validate the user's edits.</li>
+ * <li>Call {@link #applyToVariable()} to update the target variable with the
+ * new properties.</li>
+ * </ol>
+ */
+public final class EditVariable
+{
+    private final ImmutableList<VariableGroup> fAllGroups;
+
+    private final AbstractVariable fTarget;
+
+    private String fDisplayName;
+    
+    private String fHelpText;
+    
+    private int fGroupId;
+    
+    private final SortedSet<RiskModel> fDependentModels;
+    
+    /**
+     * Constructs an instance.
+     * @param variable the target variable. Will be stored, but not modified
+     * until calling {@link #applyToVariable()}.
+     * @param allGroups all available VariableGroups for user selection. Will
+     * make a defensive copy, preserving iteration order.
+     */
+    public EditVariable(
+            final AbstractVariable variable,
+            final Collection<VariableGroup> allGroups) 
+    {
+        fTarget = variable;
+        fDisplayName = variable.getDisplayName();
+        fHelpText = variable.getHelpText();
+        fGroupId = variable.getGroup().getId();
+        fAllGroups = ImmutableList.copyOf(allGroups);  // defensive copy
+        fDependentModels = new TreeSet<>();
+    }
+    
+    /**
+     * Calculates and stores which, if any, of the given RiskModels depend on
+     * the target variable.
+     * @param allModels the models to check
+     */
+    public void calculateDependentModels(final Collection<RiskModel> allModels)
+    {
+        for (final RiskModel model : allModels)
+        {
+            if (model.getRequiredVariables().contains(fTarget))
+            {
+                fDependentModels.add(model);
+            }
+        }
+    }
+    
+    /**
+     * <p>If {@link #calculateDependentModels(Collection)} has been called,
+     * returns the set of RiskModels that depend on this variable for the
+     * user's reference. Otherwise, returns an empty set.</p>
+     * @return a set sorted by the RiskModels' natural order
+     */
+    public SortedSet<RiskModel> getDependentModels()
+    {
+        return fDependentModels;
+    }
+
+    /**
+     * Returns the collection of all VariableGroups provided to the constructor.
+     * Iteration order is preserved.
+     */
+    public ImmutableCollection<VariableGroup> getAllGroups()
+    {
+        return fAllGroups;
+    }
+
+    /**
+     * The key that uniquely identifies the variable, suitable for displaying
+     * to an administrative user.
+     */
+    public String getKey()
+    {
+        // Just return the current key of the target since it's read-only.
+        return fTarget.getKey();
+    }
+
+    public String getDisplayName()
```

Comment:
Yes, it's pretty self-explanatory, but we agreed that every method should have a Javadoc so I'll add them.